src/ansys/units/unit.py

```diff
@@ -121,34 +132,243 @@ def _units_to_dim(self, units: str, power: float = None, dimensions: dict = None
+        for term, term_info in _fundamental_units.items():
+            if term_info["type"] == unit_term_type and term_info["factor"] == 1.0:
+                return term
+
+    def _condense(self, units=str) -> str:
+        """
+        Condense a unit string by collecting like terms.
+
+        Parameters
+        ----------
+        units : str
+            Unit string to simplify.
+
+        Returns
+        -------
+        str
+            Simplified unit string.
+        """
+        terms_and_powers = {}
+        units = units.strip()
+
+        # Split unit string into terms and parse data associated with individual terms
+        for term in units.split(" "):
+            _, unit_term, unit_term_power = self.filter_unit_term(term)
+
+            if unit_term in terms_and_powers:
+                terms_and_powers[unit_term] += unit_term_power
+            else:
+                terms_and_powers[unit_term] = unit_term_power
+
+        units = ""
+
+        # Concatenate unit string
+        for term, power in terms_and_powers.items():
+            if not (power):
+                continue
+            if power == 1.0:
+                units += f"{term} "
+            else:
+                power = int(power) if power % 1 == 0 else power
+                units += f"{term}^{power} "
+
+        return units.rstrip()
+
+    def filter_unit_term(self, unit_term: str) -> tuple:
+        """
+        Separate multiplier, base, and power from a unit term.
+
+        Parameters
+        ----------
+        unit_term : str
+            Unit term of the unit string.
+
+        Returns
+        -------
+        tuple
+            Tuple containing the multiplier, base, and power of the unit term.
+        """
+        multiplier = ""
+        power = 1.0
+
+        # strip power from unit term
+        if "^" in unit_term:
+            power = float(unit_term[unit_term.index("^") + 1 :])
+            unit_term = unit_term[: unit_term.index("^")]
+
+        base = unit_term
+
+        # strip multiplier and base from unit term
+        has_multiplier = self._has_multiplier(unit_term)
+        if has_multiplier:
+            for mult in _multipliers:
+                if unit_term.startswith(mult):
+                    if not self._has_multiplier(unit_term[len(mult) :]):
+                        multiplier = mult
+                        base = unit_term[len(mult) :]
+                        break
+
+        # if we thought it had a multiplier, that's just because the string wasn't
+        # a known unit on its own. So if we can't actually find its multiplier then
+        # this string is an invalid unit string
+        if has_multiplier and not multiplier:
+            raise UnitError.UNKNOWN_UNITS(unit_term)
+        return multiplier, base, power
+
+    def si_data(
+        self,
+        units: str = None,
+        power: float = None,
+        si_units: str = None,
+        si_multiplier: float = None,
+    ) -> tuple:
```

Comment:
Would maybe a data class be more appropriate here than a raw tuple?

https://docs.python.org/3/library/dataclasses.html

Reply:
Staring at this method a little more I'm a little bit confused by it.

Could you add some unit tests for it that exercises the different valid combinations of input parameters?

If I read the method implementation correctly it seems like it does not belong in this class, it has no dependencies on the encapsulated data or methods of the `Unit` instance on which it's called.  Maybe it belongs in the registry?

Dan

Reply:
Alright, stared at it some more, never mind this comment for now, let's just leave it as is.